test/app_automate_test.rb

```diff
@@ -0,0 +1,279 @@
+require 'test_helper'
+require_relative '../lib/features/support/app_automate_driver'
+require_relative '../lib/features/support/capabilities/devices'
+
+class AppAutomateTest < Test::Unit::TestCase
+
+  USERNAME = "Username"
+  ACCESS_KEY = "Access_key"
+  LOCAL_ID = "Local_id"
+  APP_LOCATION = "App_location"
+
+  def test_default_assignments
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    assert_equal(USERNAME, driver.instance_variable_get( :@username ))
+    assert_equal(ACCESS_KEY, driver.instance_variable_get( :@access_key ))
+    assert_equal(LOCAL_ID, driver.instance_variable_get( :@local_id ))
+    assert_equal(:id, driver.instance_variable_get( :@locator ))
+    assert_equal({
+      'browserstack.console': 'errors',
+      'browserstack.localIdentifier': LOCAL_ID,
+      'browserstack.local': 'true',
+      'browserstack.networkLogs': 'true'
+    }, driver.instance_variable_get( :@capabilities ))
+  end
+
+  def test_overridden_locator
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, :accessibility_id)
+    assert_equal(:accessibility_id, driver.instance_variable_get( :@locator ))
+  end
+
+  def test_devices_source
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    assert_equal(Devices::DEVICE_HASH, driver.send(:devices))
+  end
+
+  def test_click_element_nil
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    appium_mock = Minitest::Mock.new
+    driver.instance_variable_set(:@driver, appium_mock)
+
+    appium_mock.expect(:nil?, true)
+
+    driver.click_element("test_button")
+
+    appium_mock.verify
+
+  ensure
+    driver.instance_variable_set(:@driver, nil)
+  end
+
+  def test_click_element_defaults
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    appium_mock = Minitest::Mock.new
+    driver.instance_variable_set(:@driver, appium_mock)
+
+    appium_mock.expect(:nil?, false)
+    appium_mock.expect(:click, true)
+    appium_mock.expect(:find_element, appium_mock) do |locator, element_id|
+      locator === :id && element_id === "test_button"
+    end
+
+    driver.click_element("test_button")
+
+    appium_mock.verify
+
+  ensure
+    driver.instance_variable_set(:@driver, nil)
+  end
+
+  def test_click_element_locator
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, :accessibility_id)
+    appium_mock = Minitest::Mock.new
+    driver.instance_variable_set(:@driver, appium_mock)
+
+    appium_mock.expect(:nil?, false)
+    appium_mock.expect(:click, true)
+    appium_mock.expect(:find_element, appium_mock) do |locator, element_id|
```

Comment:
`find_element` should be stubbed to return a different mock, not `appium_mock` itself